pkg/controllers/provisioningv2/rke2/provisioningcluster/controller.go

```diff
@@ -181,6 +181,13 @@ func (h *handler) OnChange(_ string, cluster *rancherv1.Cluster) (*rancherv1.Clu
 		if machinePool.DynamicSchemaSpec != "" && json.Unmarshal([]byte(machinePool.DynamicSchemaSpec), &spec) == nil {
 			continue
 		}
+		if machinePool.NodeConfig == nil {
+			continue
+		}
```

Comment:
Nit: I think this change is ok, and I just want to mention that the inner loop returns an error when `NodeConfig` is nil instead of continue.